### PR TITLE
[plugins/automod] Add bad words filter

### PR DESCRIPTION
**Pull request by puxlit**
(Because I don't wanna keep rebasing this damn branch.)

### CODE REVIEW

File: src/plugins/automod/filters/BadWords.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const { remove } = require('confusables');
+const Filter = require('../structs/Filter.js');
+
+function compilePattern(pattern) {
+    return new RegExp(...(typeof(pattern) === 'string' ? [pattern, 'i'] : pattern));
+}
+
+function describeMatchedPatterns(preamble, patterns, content) {
+    let patternsDesc = patterns.filter(re => re.test(content)).map(re => `\t\u2022 \`${re}\``).join('\n');
```
Comment (puxlit) [inline src/plugins/automod/filters/BadWords.js]: Yeah yeah if the pattern contains a backtick then the description's gonna look weird.

Comment (LunarTwilight) [reply src/plugins/automod/filters/BadWords.js]: Doesn't twinkle have a function that will format out/escape backticks? (@Dorumin ?)

File: src/plugins/automod/filters/BadWords.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (this.patterns.some(re => re.test(message.content))) return true;
+
+        const ccnormContent = remove(message.content);
+        return this.postCcnormPatterns.some(re => re.test(ccnormContent));
+    }
+
+    async handle(message) {
+        const muteAction = message.member.addRole('401231955741507604');
+        message.author.send(`Hey! Watch your language! You've been grounded from ${message.guild.name}; message someone with the **@Server Moderator** role to talk this out.`); // TODO # of offenses
```
Comment (LunarTwilight) [inline src/plugins/automod/filters/BadWords.js]: Should we also include admins, or just modes?

Comment (LunarTwilight) [reply src/plugins/automod/filters/BadWords.js]: discussed in discord, admins aren't needed to be listed

File: src/plugins/automod/filters/BadWords.js (hunk 1 above)
Comment (Dorumin) [reply src/plugins/automod/filters/BadWords.js]: No need to escape backticks, as we control the input; if we ever need to, dependencies could be added to the AutoMod plugin and you can just require the fmt plugin to use its `.code` helper.

Comment (LunarTwilight) [reply src/plugins/automod/filters/BadWords.js]: I'd rather we deal with this now/soon, rather than waiting for it to come back and bite us, but eh.

Comment (puxlit) [reply src/plugins/automod/filters/BadWords.js]: For posterity: <https://discordapp.com/channels/246075715714416641/476452336282107925/672369870167474176>.

**Pull request merged**